lib/src/main/kotlin/com/lemonappdev/konsist/core/provider/KoAnnotationProviderCore.kt

```diff
@@ -15,7 +16,7 @@ internal interface KoAnnotationProviderCore :
     override val annotations: List<KoAnnotationDeclaration>
         get() = ktAnnotated
             .annotationEntries
```

Comment:
Is it correct that I used `as` instead of `as?. Am I right that we prefer the exception in this case?

Reply:
An exception should be Konsist custom exception (`KoInternalException`), not a generic one (`ClassCastException`)

ClassCastException exception
```kotlin
"ABC" as Int

Konsist Exception
```kotlin
("ABC" as? Int) ?: KoInternalException("In is null")

if(newInt == null) {

}
```